File: baekjoon/python/success/1463.py
# https://www.acmicpc.net/problem/1463


def make_one(result_set, count=0):

    if 1 in result_set:
        return count

    tmp_set = set()

    for num in result_set:

        if num % 3 == 0:
            tmp_set.add(num / 3)

        if num % 2 == 0:
            tmp_set.add(num / 2)

        tmp_set.add(num - 1)

    count += 1

    return make_one(tmp_set - result_set, count)


if __name__ == "__main__":

    input_set = {int(input())}

    print(make_one(input_set))

File: baekjoon/cpp/success/1463.cpp
// https://www.acmicpc.net/problem/1463

#include <iostream>
#include <set>

using namespace std;

int makeOne(set<int> &inputSet, int cnt) {

    set<int>::iterator iter;
    for (iter = inputSet.begin(); iter != inputSet.end(); ++iter) {

        if (*iter == 1) {
            return cnt;
        }
    }

    set<int> tmpSet;
    for (iter = inputSet.begin(); iter != inputSet.end(); ++iter) {

        if (*iter % 3 == 0) {
            tmpSet.insert(*iter / 3);
        }

        if (*iter % 2 == 0) {
            tmpSet.insert(*iter / 2);
        }

        tmpSet.insert(*iter - 1);
    }

    cnt += 1;

    for (iter = inputSet.begin(); iter != inputSet.end(); ++iter) {

        tmpSet.erase(*iter);
    }

    return makeOne(tmpSet, cnt);
}

int main(){

    set<int> inputSet;

    int inputInt = 0;
    cin >> inputInt;

    inputSet.insert(inputInt);

    printf("%d\n", makeOne(inputSet, 0));

    return 0;
}